octodns/provider/googlecloud.py

```diff
@@ -0,0 +1,363 @@
+#
+#
+#
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+import re
+import shlex
+import time
+from logging import getLogger
+
+from google.cloud import dns
+
+from .base import BaseProvider
+from ..record import Record
+
+
+class _GoogleCloudRecordSetMaker(object):
+    """Wrapper to make google cloud client resource record sets from OctoDNS
+    Records.
+
+        googlecloud.py:
+        class: octodns.provider.googlecloud._GoogleCloudRecordSetMaker
+        An _GoogleCloudRecordSetMaker creates google cloued client resource
+        records which can be used to update the Google Cloud DNS zones.
+    """
+
+    def __init__(self, gcloud_zone, record):
+        self.gcloud_zone = gcloud_zone
+        self.record = record
+
+        self._record_set_func = getattr(
+            self, '_record_set_from_{}'.format(record._type))
+
+    def get_record_set(self):
+        return self._record_set_func(self.record)
+
+    def _record_set_from_A(self, record):
+        return self.gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, record.values)
+
+    _record_set_from_AAAA = _record_set_from_A
+
+    def _record_set_from_CAA(self, record):
+        return self.gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, [
+                '{flags} {tag} {value}'.format(**record.data['value'])])
+
+    def _record_set_from_CNAME(self, record):
+        return self.gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, [record.value])
+
+    def _record_set_from_MX(self, record):
+        return self.gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, [
+                '{preference} {exchange}'.format(**v.data)
+                for v in record.values])
+
+    def _record_set_from_NAPTR(self, record):
+        return self.gcloud_zone.resource_record_set(
+            record.fqdn, record._type, record.ttl, [
+                '{order} {preference} "{flags}" "{service}" '
+                '"{regexp}" {replacement}'
+                .format(**v.data) for v in record.values])
+
+    _record_set_from_NS = _record_set_from_A
+
+    _record_set_from_PTR = _record_set_from_CNAME
+
+    _record_set_from_SPF = _record_set_from_A
```

Comment:
In other providers TXT and SPF records use the same implementation as they're relatively similar records. Would that make sense here or is there a specific reason it's alised to A?

Reply:
yes. Updated.